client/src/lib.rs

```diff
@@ -360,10 +362,13 @@ impl State {
                 player_state,
                 dt,
             );
+
+            // update player instance (replace 0 with player_id) (maybe move to scene graph later)
+            let player_instances = self.scene.objects_and_instances.get_mut(&scene::ModelIndex{index: scene::ModelIndices::PLAYER as usize}).unwrap();
+            player_instances[0].transform = self.player.calc_transf_matrix();
```

Comment:
> The childtransform and the query issues are fixed: you can query a node with a node id and get the node and its transforms based on child alone. Let me know what you think, and whether you have any other suggestions.

Nice! With the hashmap we can change this to a hashmap query right?

Reply:
Everything else looks good. Right now I see that NodeIDs are fixed, but later when adding multiple player node we can easily extend that in a dynamic way.

Reply:
>Nice! With the hashmap we can change this to a hashmap query right?
Yes! But then you would have to call scene.draw_scene_dfs() to get the updated objects_and_instances before you render the scene

Reply:
if `draw_scene_dfs` is not too costly, it can be called before every render right? So we effectively bind the scene graph to the instances that are actually to be rendered

Reply:
Yea exactly. draw_scene_dfs shouldn't be too costly I think.

Reply:
ok everything makes sense to me then 👍